CreateItem/CreateItemWrapper.js

```diff
@@ -1,78 +1,77 @@
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import PropTypes from 'prop-types';
-import { Button, Icon, omitProps } from '@folio/stripes/components';
 import className from 'classnames';
 
+import {
+  Button,
+  Icon,
+} from '@folio/stripes/components';
+
 import css from './CreateItemWrapper.css';
 import CreateItemModal from './CreateItemModal';
 
-export default class CreateItemWrapper extends React.Component {
-  constructor(props) {
-    super(props);
-
-    this.connectedCreateItemModal = props.stripes.connect(CreateItemModal, { dataKey: props.dataKey });
-  }
-
-  state = {
-    openModal: false,
-  }
-
-  getStyle() {
-    const { marginBottom0, marginTop0 } = this.props;
+const CreateItemWrapper = ({
+  addItem,
+  disabled,
+  instanceId,
+  locationId,
+  marginBottom0,
+  marginTop0,
+  searchButtonStyle,
+  searchLabel,
+}) => {
+  const [isModalOpened, setIsModalOpened] = useState(false);
 
-    return className(
-      css.searchControl,
-      { [css.marginBottom0]: marginBottom0 },
-      { [css.marginTop0]: marginTop0 },
-    );
-  }
+  const toggleModal = () => setIsModalOpened(!isModalOpened);
 
-  openModal = () => this.setState({
-    openModal: true,
-  });
+  const getStyle = useMemo(
```

Comment:
I'm not telling you to remove `useMemo` here, but there is no reason to use it in this particular case. It was invented to cache result of "heavy" calculations, or API callbacks. In cases like here usually it has no impact or even negative impact on memory consumption and so on. Maybe it's a good idea to look for some thoughts in articles, like this for example https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Thanks. it could be removed